capstone-project/src/main/angular-webapp/src/app/map/map.component.ts

```diff
@@ -42,6 +45,41 @@ export class MapComponent implements OnInit {
       });
   }
 
+  // Centers the map based on the user location if permission is granted.
+  focusOnUserLocation() {
+
+    // Browser supports Geolocation
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition((position) => {
+          const pos = {
+            lat: position.coords.latitude,
+            lng: position.coords.longitude
+          };
+          this.gMap.setCenter(pos);
+          this.gMap.setZoom(14);
+        },
+        () => {
+          MapComponent.handleLocationError(true);
+        }
+      );
+    }
+
+    // Browser doesn't support Geolocation
+    else {
+      MapComponent.handleLocationError(false);
+    }
+  }
+
+  // Alerts the user if the location process failed.
+  static handleLocationError(broswerHasGeolocation: boolean) {
+    if (broswerHasGeolocation) {
+      window.alert("Geolocation service failed. Please grant the browser permission to locate you.")
+    }
+    else {
+      window.alert("Browser doesn't support Geolocation.")
+    }
+  }
```

Comment:
Alerts are nice but very primitive UX :) 

Read about Toast service (you can do in following PR)